.github/workflows/tests.yml

```diff
@@ -0,0 +1,61 @@
+name: tests
+
+on:
+  push:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  tests:
+    name: Test
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python_version: ["3.8", "3.9", "3.10", "3.11"]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v3
+
+      - name: Configure Python version
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python_version }}
+          architecture: x64
+          cache: "pip"
+
+      - name: Build sdist
+        run: |
+          python -m pip install -U build pip setuptools
+          python -m pip install -U -r requirements.txt
+          python -m build --sdist
+
+      - name: Run mypy
+        run: python -m mypy catalogue
+
+      - name: Delete source directory
+        uses: JesseTG/rm@v1.0.3
```

Comment:
It feels weird to depend on some random person's action here. Is there no built-in option?

Reply:
I followed what Ines was doing here for standardization reason: https://github.com/explosion/radicli/blob/main/.github/workflows/test.yml#L26-L29
I can look into something else if you prefer.